frontend/src/components/contacts.tsx

```diff
@@ -0,0 +1,275 @@
+      )
+    );
+  };
+
+  const handleRemoveImportantEvent = (index: number) => {
+    setImportantDates((prevDates) => [
+      ...prevDates.slice(0, index),
+      ...prevDates.slice(index + 1),
+    ]);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      await axios.post("http://localhost:8000/user/contacts", {
+        firstName,
+        lastName,
+        relationships,
+        importantDates,
+        email,
+        phoneNumber,
+        notes,
+      });
+    } catch (error) {
+      console.error(error);
+      setErrorMessage((error as any).response.date ?? "Unknown error occured.");
+    }
+  };
+
+  return (
+    <div className="contacts_form">
+      <h1>Add a contact</h1>
+      <form onSubmit={handleSubmit}>
+        {errorMessage && <p className="error_message">{errorMessage}</p>}
+        <div className="user_input">
+          <label htmlFor="first_name">First name</label>
+          <input
+            className="first_name"
+            name="first_name"
+            type="text"
+            autoComplete="on"
+            required
+            value={firstName}
+            onChange={handleFirstNameChange}
+          ></input>
+
+          <label htmlFor="last_name">Last name</label>
+          <input
+            className="last_name"
+            name="last_name"
+            type="text"
+            autoComplete="on"
+            required
+            value={lastName}
+            onChange={handleLastNameChange}
+          ></input>
+
+          <label htmlFor="email">Email</label>
+          <input
+            className="email"
+            name="email"
+            type="text"
+            autoComplete="on"
+            required
+            value={email}
+            onChange={handleEmailChange}
+          ></input>
+
+          <label htmlFor="phone_number">Phone number</label>
+          <input
+            className="phone_number"
+            name="phone_number"
+            type="text"
+            autoComplete="on"
+            required
+            value={phoneNumber}
+            onChange={handlePhoneNumberChange}
+          ></input>
+
+          <label htmlFor="notes">Extra Notes</label>
+          <input
+            className="notes"
+            name="notes"
+            type="text"
+            autoComplete="on"
+            required
+            value={notes}
+            onChange={handleNotesChange}
+          ></input>
+
+          <label>
+            Relationship with Person
+            <input
+              type="text"
+              value={relationshipType}
+              onChange={(event) => {
+                const value = event.target.value;
+                if (/^[A-Za-z\s]*$/.test(value)) {
+                  setRelationshipType(value);
+                }
+              }}
+            />
+          </label>
+          <button onClick={addRelationship}>Add</button>
```

Comment:
The `add` button for relationship and date/event is causing the form to submit which runs the handleSubmit function. We'd want to add a `type="button"` here 
```suggestion
          <button type="button" onClick={addRelationship}>Add</button>
```

Reply:
Would this also apply to the remove relationship button? For example
 <button onClick={() => handleRemoveImportantEvent(index)}>
          x
 </button>

Reply:
Good to know will change this!